static/js/admin.js

```diff
@@ -1026,6 +1030,35 @@ function _setup_page() {
         });
     });
 
+    function upload_realm_icon(file_input) {
+        var form_data = new FormData();
+
+        form_data.append('csrfmiddlewaretoken', csrf_token);
+        jQuery.each(file_input[0].files, function (i, file) {
+            form_data.append('file-'+i, file);
+        });
+
+        var spinner = $("#upload_icon_spinner").expectOne();
+        loading.make_indicator(spinner, {text: 'Uploading icon.'});
+
+        channel.put({
+            url: '/json/realm/icon',
+            data: form_data,
+            cache: false,
+            processData: false,
+            contentType: false,
+            success: function (data) {
+                loading.destroy_indicator($("#upload_icon_spinner"));
+                var url = data.icon_url + '&stamp=' + exports.icon_stamp;
+                $("#realm-settings-icon").expectOne().attr("src", url);
+                $("#realm_icon_delete_button").show();
+                exports.icon_stamp += 1;
```

Comment:
(Also, do we need the `icon_stamp` code here?  I would think that'd be handled by version)